control/position_controller.hpp

```diff
@@ -0,0 +1,177 @@
+/*******************************************************************************
+ * Copyright (c) 2009-2016, MAV'RIC Development Team
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ ******************************************************************************/
+
+/*******************************************************************************
+ * \file position_controller.hpp
+ *
+ * \author MAV'RIC Team
+ * \author Basil Huber
+ *
+ * \brief Interface for position controller
+ *
+ ******************************************************************************/
+
+
+#ifndef POSITION_CONTROLLER_HPP_
+#define POSITION_CONTROLLER_HPP_
+
+#include "control/iposition_controller.hpp"
+#include "control/ixyposition_zvel_controller.hpp"
+#include "control/inavigation_controller.hpp"
+#include "control/pid_controller.hpp"
+#include "control/stabilisation.hpp"
+#include "sensing/ins.hpp"
+#include "sensing/ahrs.hpp"
+
+class Position_controller : public IPosition_controller, public IXyposition_zvel_controller, public INavigation_controller
+{
+public:
+
+    struct conf_t
+    {
+        pid_controller_conf_t cruise_pid_config;
+        pid_controller_conf_t hover_pid_config;
+        float max_climb_rate;
+        float max_rel_yaw;
+        float min_cruise_dist;
+        float kp_yaw;
+    };
+
+    enum class ctrl_mode_t
+    {
+        POS_XYZ,
+        POS_XY_VEL_Z
+    };
+
+    Position_controller(INS& ins, ahrs_t& ahrs, conf_t config);
```

Comment:
done in https://github.com/lis-epfl/MAVRIC_Library/pull/331/commits/242e95ef6cfece280ea34ff986bfb9dd6c95f5a4